src/pal/pal_linux.h

```diff
@@ -26,6 +25,16 @@ namespace snmalloc
      */
     static constexpr uint64_t pal_features = PALPOSIX::pal_features;
 
+#  ifdef SNMALLOC_OS_PAGE_SIZE
+    /*
+     * Allow build-time customization of OS_PAGE_SIZE since Linux is apparently
+     * a little less opinionated across its various ports than many other OSes.
+     */
+    static constexpr size_t OS_PAGE_SIZE = SNMALLOC_OS_PAGE_SIZE;
```

Comment:
Do we want to allow a global override and assert that it's at least equal to the OS's limit?  We can always support a page size that's greater than the OS wants, just not one that's smaller.

I also wonder if we want to have a page size in the AAL and the PAL and take the max of the two.

Reply:
Do any architectures out in the wild not *support* 4K pages, at least in principle?  AFAIK ppc64 does, it just happens that Debian/ppc64le builds their kernel with 64K pages.

Reply:
I think SPARC supports only 8KiB pages (though they might have fixed that in later SPARC)?  That said, I don't particularly want to support SPARC.  

Why is the external override for Linux?  I think that I'd prefer to structure this as:

 - Add an `enum` for the architecture.
 - Make the AAL expose a constexpr field with the correct enum value.
 - Make this a `std::conditional` keyed off that field.
 - Use `SNMALLOC_OS_PAGE_SIZE` in the global constant that we currently initialise with the version from the PAL.
 - Add a static assert that this global is a multiple of `PAL::OS_PAGE_SIZE`

Does that make sense?

Reply:
Ah, indeed, UltraSPARC sayeth:

> The MMUs may support up to eight page sizes: 8 KBytes, 64 KBytes, 512 KBytes, 4 MBytes, 32 MBytes, 256 MBytes, 2 GBytes, and 16 GBytes.

When you say "Add an `enum` for the architecture" do you mean something like
```
enum aal_page_size {
  AAL_PAGE_SIZE_4K,
  // ...
  AAL_PAGE_SIZE_64K
};
```
or something else?

Reply:
I meant replace / augment the `PLATFORM_IS_X86` and similar with something like:
```c
enum Architecture
{
    X86,
    ARM,
    PowerPC
};
```

And then be able to write something like this, rather than relying on `#ifdef`s:

```c++
static constexpr size_t OS_PAGE_SIZE = AAL::Arch == Architecture::PowerPC ? 0x10000 : 0x1000;
```

Though, actually, it would make me even happier to have these in one of the helper headers:

```c++
constexpr unsigned long long int operator "" _KiB(unsigned long long int v)
{
        return v * 1024;
}

constexpr unsigned long long int operator "" _MiB(unsigned long long int v)
{
        return v * 1024 * 1024;
}
```
And then to write it as:

```c++
static constexpr size_t OS_PAGE_SIZE = AAL::Arch == Architecture::PowerPC ? 64_KiB : 4_KiB;
```

But that's not necessary for this PR unless you feel like it.